models/MarioInventory.py

```diff
@@ -1,30 +1,122 @@
 """
-This module represents Mario as an abstract object for simulating world traversal.
-This is required for checking randomization logic.
+This module represents Mario's inventory as an abstract object for simulating
+world traversal. This is required for checking randomization logic.
 """
 
 from metadata.partners_meta import all_partners
 
-class Mario:
+class MarioInventory:
     """
-    Represents a state of Mario, including items, partners and more abstract
-    things that influence progression.
+    Represents a state of Mario's inventory, including items, partners and more
+    abstract things that influence progression.
     """
-    def __init__(self, **kwargs):
-        self.boots = kwargs.get("boots", set())
-        self.hammer = kwargs.get("hammer", set())
-        self.items = kwargs.get("items", set())
-        self.starpieces = kwargs.get("starpieces", set())
-        self.partners = kwargs.get("partners", set())
-        self.favors = kwargs.get("favors", set()) # https://www.mariowiki.com/Koopa_Koot%27s_favors
-        self.flags = kwargs.get("flags", set())
-        self.starspirits = kwargs.get("starspirits", set())
-        self.item_history = []
+    def __init__(
+        self,
+        starting_boots:int=0,
+        starting_hammer:int=0,
+        starting_partners:list=None,
+        starting_items:list=None,
+        partners_always_usable:bool=True,
+        hidden_block_mode:int=1,
+        magical_seeds_required:int=4,
+        startwith_bluehouse_open:bool=False,
+        startwith_toybox_open:bool=False,
+        startwith_whale_open:bool=False,
+        startwith_speedyspin:bool=True,
+        cook_without_fryingpan:bool=True
+    ):
+        """
+        Initializes Mario's starting inventory with values depending on
+        several parameters and flags.
+        This includes partners (for considering their overworld abilities during
+        world graph traversal), equipment, as well as additional pseudoitems
+        based on chosen settings.
+        Starting equipment irrelevant to world graph traversal (such as Lucky
+        Star, starting coins) is ignored.
+        """
+        assert(isinstance(starting_boots, int) and starting_boots in range(-1,3))
```

Comment:
Shouldn't the range be (-1,2)?

Reply:
python's range function includes the first, but excludes the second parameter. so range(-1,3) produces [-1,0,1,2]

Reply:
I always forget this weirdness, great then